lib/rmt/cli/products.rb

```diff
@@ -21,10 +21,15 @@ def list
     decorator = ::RMT::CLI::Decorators::ProductDecorator.new(products)
 
     if products.empty?
+      no_match_prod_db_message = 'No matching products found in the database.'
       if options.all
-        warn _("Run '%{command}' to synchronize with your SUSE Customer Center data first.") % { command: 'rmt-cli sync' }
+        if options.name.blank? && options.version.blank? && options.arch.blank?
+          warn _("Run '%{command}' to synchronize with your SUSE Customer Center data first.") % { command: 'rmt-cli sync' }
+        else
+          warn _(no_match_prod_db_message)
+        end
       else
-        warn _('No matching products found in the database.')
+        warn _(no_match_prod_db_message)
```

Comment:
I think this could be expressed without a nested if-statement. Something like:
```ruby
if options.all && options.name.blank? && options.version.blank? && options.arch.blank?
  warn _("Run '%{command}' to synchronize with your SUSE Customer Center data first.") % { command: 'rmt-cli sync' }
else
  warn _('No matching products found in the database.')

Reply:
I think I tried that and gave me an error, probably a typo and didnt see it. Done now.